torchtune/utils/_device.py

```diff
@@ -156,3 +181,71 @@ def batch_to_device(batch: dict, device: torch.device) -> None:
                 f"""To use batch_to_device, all elements in the batch must be a dict or Tensor.
 Got key "{k}" with value of type {type(v)}"""
             )
+
+
+class DeviceSupport(Enum):
+    """
+    This is a simple enum for compute devices,
+    This currently only supports CPU, CUDA, NPU.
+    The following enumeration defines various device configurations with attributes:
+    1. `device_type` (str): The type of device (e.g., "cpu", "cuda", "npu").
+    2. `device_name` (str): A user-friendly name for the device (e.g., "CPU", "GPU", "NPU").
+    3. `communication_backend` (str): Specifies the backend used for communication on this device (e.g., "gloo", "nccl", "hccl").
+    4. `cuda_like` (bool): Indicates whether the device is CUDA-like or not.
```

Comment:
Discussed in another comment but I'm not sure we 100% need this field. At the very least I find its naming a bit unclear

Reply:
Thank you, I’ll remove this seemingly redundant attribute!